core/vm/logger.go

```diff
@@ -267,6 +269,18 @@ func (l *StructLogger) CaptureState(pc uint64, op OpCode, gas, cost uint64, scop
 		if err := traceStorage(l, scope, structlog.getOrInitExtraData()); err != nil {
 			log.Error("Failed to trace data", "opcode", op.String(), "err", err)
 		}
+
+		oldValue := l.env.StateDB.GetState(contractAddress, storageKey)
+		if !bytes.Equal(oldValue.Bytes(), storageValue.Bytes()) &&
+			bytes.Equal(storageValue.Bytes(), common.Hash{}.Bytes()) {
+
+			if delSibling, err := l.env.StateDB.GetDeletetionProof(contractAddress, storageKey); err != nil {
```

Comment:
What's the expected performance impact of this?

Reply:
Okay, sounds acceptable. We just need to think if this can be triggered maliciously somehow.

Reply:
You are right. I just recognized that current implement would bring an addition commition on the corresponding storage trie so it may bring significant cost by a series well designed malicious txs in one block.

Lucikly if we should have separated and constrainted the logtrace only work while obtaining block trace, such a attack is mitigated for the cost only raised in any following nodes used for providng block trace.

And I think we can further push the deletion generation process to commiting step, I would start this idea in another PR.

Reply:
Would trace this problem in https://github.com/scroll-tech/go-ethereum/issues/253